src/tools/immutable_object_cache/ObjectCacheStore.cc

```diff
@@ -193,11 +243,28 @@ int ObjectCacheStore::lookup_object(std::string pool_nspace,
 
   switch (ret) {
     case OBJ_CACHE_NONE: {
-      pret = do_promote(pool_nspace, pool_id, snap_id, object_name);
-      if (pret < 0) {
-        lderr(m_cct) << "fail to start promote" << dendl;
+      if (take_token_from_throttle(object_size, 1)) {
+        pret = do_promote(pool_nspace, pool_id, snap_id, object_name);
+        if (pret < 0) {
+          lderr(m_cct) << "fail to start promote" << dendl;
+        }
+        return ret;
+      } else {
+        m_policy->update_status(cache_file_name, OBJ_CACHE_THROTTLE);
+        return OBJ_CACHE_THROTTLE;
+      }
+    }
+    case OBJ_CACHE_THROTTLE: {
+      if (take_token_from_throttle(object_size, 1)) {
+        m_policy->update_status(cache_file_name, OBJ_CACHE_SKIP);
+        pret = do_promote(pool_nspace, pool_id, snap_id, object_name);
+        if (pret < 0) {
+          lderr(m_cct) << "fail to start promote" << dendl;
+        }
+        return OBJ_CACHE_SKIP;
+      } else {
+        return ret;
```

Comment:
@dillaman I deleted the thread method, still judge the throttle here. The difference is that the status is changed according to the throttle and a new state is introduced "OBJ_CACHE_THROTTLE"